frontend/src/pages/RegisterPage.tsx

```diff
@@ -0,0 +1,125 @@
+          <div className="relative mt-6 py-6">
+            <input
+              type="email"
+              id="email"
+              name="email"
+              placeholder=""
+              className="peer absolute top-0 left-0 w-full border-2 rounded-xl p-2 border-slate-100 focus:outline-none focus:border-2 focus:border-sky-300"
+              required
+            />
+            <label
+              htmlFor="email"
+              className="transition-all absolute p-2 top-0 left-0 z-10 select-none peer-focus:text-sky-500 peer-focus:-top-8 peer-focus:text-sm peer-[:not(:placeholder-shown)]:-top-8 peer-[:not(:placeholder-shown)]:text-sm"
+            >
+              Email
+            </label>
+          </div>
+          <div className="relative mt-6 py-6">
+            <input
+              type="password"
+              id="password"
+              name="password"
+              placeholder=""
+              className={
+                "peer absolute top-0 left-0 w-full border-2 rounded-xl p-2 border-slate-100 focus:outline-none focus:border-2 focus:border-sky-300 " +
+                (error.id === 1 || error.id === 2 ? "border-red-300" : "")
+              }
+              required
+            />
+            <label
+              htmlFor="password"
+              className="transition-all absolute p-2 top-0 left-0 z-10 select-none peer-focus:text-sky-500 peer-focus:-top-8 peer-focus:text-sm peer-[:not(:placeholder-shown)]:-top-8 peer-[:not(:placeholder-shown)]:text-sm"
+            >
+              Hasło
+            </label>
+            <p
+              className={
+                "absolute p-2 -top-8 right-0 z-10 text-sm text-red-500 " +
+                (error.id === 1 || error.id === 2 ? "" : "hidden")
+              }
+            >
+              {error.message}
+            </p>
+          </div>
+          <div className="relative mt-6 py-6">
+            <input
+              type="password"
+              id="repeat-password"
+              name="repeat-password"
+              placeholder=""
+              className={
+                "peer absolute top-0 left-0 w-full border-2 rounded-xl p-2 border-slate-100 focus:outline-none focus:border-2 focus:border-sky-300 " +
+                (error?.id === 1 ? "border-red-300" : "")
+              }
+              required
+            />
+            <label
+              htmlFor="repeat-password"
+              className="transition-all absolute p-2 top-0 left-0 z-10 select-none peer-focus:text-sky-500 peer-focus:-top-8 peer-focus:text-sm peer-[:not(:placeholder-shown)]:-top-8 peer-[:not(:placeholder-shown)]:text-sm"
+            >
+              Powtórz hasło
+            </label>
+            <p
+              className={
+                "absolute p-2 -top-8 right-0 z-10 text-sm text-red-500 " +
+                (error.id === 1 ? "" : "hidden")
+              }
+            >
+              {error.message}
+            </p>
```

Comment:
Ta sama sytuacja co wcześniej dodaj if statement.

```suggestion
            {error?.type === "NO_MATCH" && <p
              className="absolute p-2 -top-8 right-0 z-10 text-sm text-red-500">
              {error.message}
            </p>}
```